doc/jinja-filters.md

```diff
@@ -0,0 +1,438 @@
+("2018-08-14", "d MMMM YYYY")       "<span class='date'>14 August 2018</span>"
+```
+
+## format_conditional_date
+This is to be used when passing multiple dates as multiple arguments. It will return the first valid date formatted.
+
+##### Parameters: 
+- *dates: Multiple date values
+
+##### Context:
+If a user decides to report for different dates then following questions should use the user entered date. Otherwise continue asking for the metadata period.
+```
+For the period {{ format_conditional_date (answers['period-from'], metadata['ref_p_start_date'])}} to {{ format_conditional_date (answers['period-to'], metadata['ref_p_end_date'])}}, what was the value of {{ first_non_empty_item(metadata['trad_as'], metadata['ru_name']) }}’s <em>total retail turnover</em>?
+```
+
+##### Examples:
+```
+Input:                              Output:
+("2016-01-12", "2016-02-12")        "<span class='date'>12 January 2016</span>"
+("2017-12-23", None)                "<span class='date'>23 December 2017</span>"
+(None, "2017-12-24")                "<span class='date'>24 December 2017</span>"
+```
+
+## calculate_offset_from_weekday_in_last_whole_week
+Offsets a date from a particular day of the week in the previous week. 
+Intended to be used to generate reference date ranges around a particular date
+The offsets will always be based on one day of the week in the previous Mon-Sun
+
+##### Parameters: 
+- input_datetime: The datetime (or date) to offset. Defaults to today's date
+- offset: Dictionary of the number of days, weeks, or years to offset by as integer values
+- day_of_week: The day of the previous week to offset from (two letter abbreviation). Default 'MO'
+
+##### Context:
+Used in LMS for both Questions and Options. 
+Used for LMS as they would like reference dates that are more relevant to time you're filling in the survey.
+Often combined with format_date_custom filter.
+```
+If you had been offered a job in the week starting {{ calculate_offset_from_weekday_in_last_whole_week(collection_metadata['started_at'], {}) | format_date_custom( 'EEEE d MMMM' ) }} would you be able to start before {{ calculate_offset_from_weekday_in_last_whole_week(collection_metadata['started_at'], {'weeks':2}) | format_date_custom( 'EEEE d MMMM' ) }}?
+
+As an option:
+{{calculate_offset_from_weekday_in_last_whole_week(collection_metadata['started_at'], {}, 'MO') | format_date_custom('EEEE dd MMMM') }}
+```
+
+##### Examples:
+```
+Input:                              Output:
+("2018-08-10", {}, "SU")            "2018-08-05"  # Friday outputs previous Sunday
+("2018-08-10", {}, "FR")            "2018-08-03"  # Friday outputs previous Friday
+("2018-08-10", {"years": 1}, "FR")  "2019-08-03"  # Friday outputs previous Friday + 1 year
+
+("2018-08-05", {}, "SU")            "2018-07-29"  # Sunday outputs previous Sunday (Must be a full Sunday)
+("2018-08-05", {"weeks": 1}, "SU")  "2018-08-05"  # Previous sunday with +1 month offset, back to input
```

Comment:
Should this be `+1 week`?

Reply:
Yes but since I copied this from your test_jinja_filters test I blame you. I'll update both now.